### PR TITLE
docs: Correct ApplicationSet (spec.preservedFields)

### PR DESCRIPTION
**Pull request by jyoungs**
ApplicationSet -> spec.preservedFields is pretty well documented [here](https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Controlling-Resource-Modification/), but it's documented in the [ApplicationSet Specification](https://argo-cd.readthedocs.io/en/latest/operator-manual/applicationset/applicationset-specification/#applicationset-specification) as metadata.preservedFields.

Correcting the operator-manual to match what actually works.

FWIW, I'm not even using this now, but I know for certain it does not validate as written here. 

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [x] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

File: docs/operator-manual/applicationset.yaml [hunk 1]
```diff
@@ -214,6 +204,12 @@ spec:
                 - env-prod
           maxUpdate: 10%    # maxUpdate supports both integer and percentage string values (rounds down, but floored at 1 Application for >0%)
```
Comment (nitishfy) [inline docs/operator-manual/applicationset.yaml:206]:
```suggestion
# To preserve this annotation and label we can use the preservedFields property 
```

No harm adding this comment too, WDYT?

**Review by nitishfy — APPROVED** [review]
A minor comment but overall LGTM!

File: docs/operator-manual/applicationset.yaml [hunk 2]
```diff
@@ -178,14 +168,6 @@ spec:
     # Prevent an Application's child resources from being deleted, when the parent Application is deleted
     preserveResourcesOnDeletion: true
 
-    # which fields of the ApplicationSet should be ignored when comparing Applications.
-    ignoreApplicationDifferences:
```
Comment (jyoungs) [inline docs/operator-manual/applicationset.yaml:182]: Validation error from this:
```
decoding error: unknown field "spec.syncPolicy.ignoreApplicationDifferences"
```
Note: `ignoreApplicationDifferences` already appears below where it actually belongs

File: docs/operator-manual/applicationset.yaml (hunk 1 above)
Comment (jyoungs) [reply docs/operator-manual/applicationset.yaml:198]: I don't really think this comment adds anything of value, but I did improve the comments here a bit.

File: docs/operator-manual/applicationset.yaml (hunk 2 above)
Comment (todaywasawesome) [reply docs/operator-manual/applicationset.yaml:182]: Good catch, after this is cleaned up I think we'll be ready to merge.

**Review by todaywasawesome — CHANGES_REQUESTED** [review]

File: docs/operator-manual/applicationset.yaml (hunk 2 above)
Comment (jyoungs) [reply docs/operator-manual/applicationset.yaml:182]: I don't understand. It is cleaned up (exists where it should below and not here). WDYM?

File: docs/operator-manual/applicationset.yaml [hunk 3]
```diff
@@ -168,29 +158,19 @@ spec:
...
     # Prevents ApplicationSet controller from deleting Applications. Update is allowed
-  # applicationsSync: create-update
+    # applicationsSync: create-update
 
     # Prevents ApplicationSet controller from modifying Applications. Delete is allowed.
-  # applicationsSync: create-delete
+    # applicationsSync: create-delete
 
     syncOptions:
     - CreateNamespace=true
```
Comment (crenshaw-dev) [inline docs/operator-manual/applicationset.yaml:167]: I think this should also be deleted

**Review by crenshaw-dev — APPROVED** [review]
Thanks!

**Comment by crenshaw-dev** [comment]
@todaywasawesome can you confirm if your concerns were addressed?

**Review by todaywasawesome — APPROVED** [review]
Ready to go

**Pull request merged**